java-spiffe-core/src/main/java/io/spiffe/spiffeid/TrustDomain.java

```diff
@@ -67,10 +69,23 @@ public String toString() {
         return name;
     }
 
+    /**
+     * Returns the trust domain as SPIFFE ID string (e.g. 'spiffe://example.org')
+     *
+     * @return a String formatted as a SPIFFE ID
+     */
+    public String toIdString() {
+        return String.format("%s://%s", SpiffeId.SPIFFE_SCHEME, name);
+    }
+
     private static void validateHost(final String host) {
         if (StringUtils.isBlank(host)) {
             throw new IllegalArgumentException("Trust domain cannot be empty");
         }
+
+        if (host.length() > TRUST_DOMAIN_MAXIMUM_LENGTH) {
+            throw new IllegalArgumentException(String.format("Trust domain maximum length is %s bytes", TRUST_DOMAIN_MAXIMUM_LENGTH));
```

Comment:
minor performance nit, the exception message can actually be a const field of this class since the format parameter is a const